threescale/src/cache.rs

```diff
@@ -0,0 +1,38 @@
+use std::time::Duration;
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+pub enum Period {
+  Minute,
+  Hour,
+  Day,
+  Week,
+  Month,
+  Year,
+  Eternity,
+}
+
+#[allow(dead_code)]
+pub struct PeriodWindow {
+  start: Duration,
+  end: Duration,
+  window_type: Period,
+}
+
+#[allow(dead_code)]
+pub struct UsageReport {
+  period_window: PeriodWindow,
+  current_value: u32,
+  max_value: u32,
+}
+
+#[allow(dead_code)]
+pub struct Application<'a> {
+  app_id: String,
+  service_id: String,
+  timestamp: Duration,
+  local_state: RefCell<HashMap<&'a str,UsageReport>>,
+  remote_state: RefCell<HashMap<&'a str,UsageReport>>,
```

Comment:
This reminds me a lot about the Go cache, which makes this distinction - I'm not sure the approach in that cache using these two representations of state is the best start here - I think it is complex and a simpler way to deal with this might exist, perhaps we just need to think about the cache workflow and see whether we really do need these two maps.

Reply:
Just changed to one map that counts left hits instead of tracking remote and local states individually.